src/canvas/dialogs/dd_dialog.rs

```diff
@@ -58,6 +66,214 @@ impl KillDialog for Painter {
+            "31: SYS",
+            "34: RTMIN",
+            "35: RTMIN+1",
+            "36: RTMIN+2",
+            "37: RTMIN+3",
+            "38: RTMIN+4",
+            "39: RTMIN+5",
+            "40: RTMIN+6",
+            "41: RTMIN+7",
+            "42: RTMIN+8",
+            "43: RTMIN+9",
+            "44: RTMIN+10",
+            "45: RTMIN+11",
+            "46: RTMIN+12",
+            "47: RTMIN+13",
+            "48: RTMIN+14",
+            "49: RTMIN+15",
+            "50: RTMAX-14",
+            "51: RTMAX-13",
+            "52: RTMAX-12",
+            "53: RTMAX-11",
+            "54: RTMAX-10",
+            "55: RTMAX-9",
+            "56: RTMAX-8",
+            "57: RTMAX-7",
+            "58: RTMAX-6",
+            "59: RTMAX-5",
+            "60: RTMAX-4",
+            "61: RTMAX-3",
+            "62: RTMAX-2",
+            "63: RTMAX-1",
+            "64: RTMAX",
+        ];
+
+        let button_rect = Layout::default()
+            .direction(Direction::Horizontal)
+            .margin(1)
+            .constraints(
+                [
+                    Constraint::Length((button_draw_loc.width - 14) / 2),
+                    Constraint::Min(0),
+                    Constraint::Length((button_draw_loc.width - 14) / 2),
+                ]
+                .as_ref(),
+            )
+            .split(*button_draw_loc)[1];
+
+        let mut selected = match app_state.delete_dialog_state.selected_signal {
+            KillSignal::CANCEL => 0,
+            KillSignal::KILL(signal) => signal,
+        };
+        // 32+33 are skipped
+        if selected > 31 {
+            selected -= 2;
+        }
+
+        let layout = Layout::default()
+            .direction(Direction::Vertical)
+            .constraints(
+                vec![Constraint::Min(1)]
+                    .into_iter()
+                    .cycle()
+                    .take((button_rect.height).into())
+                    .collect::<Vec<Constraint>>()
+                    .as_ref(),
+            )
+            .split(button_rect);
+
+        let scroll_offset: usize = if selected < (layout.len() as usize) - 1 {
+            0
+        } else {
+            selected
+                + if selected == signal_text.len() - 1 {
+                    // don't scroll down if we are on the last element
+                    1
+                } else {
+                    // always show one more element
+                    2
+                }
+                - (layout.len() as usize)
+        };
+
+        let mut buttons = signal_text
+            [scroll_offset + 1..min((layout.len() as usize) + scroll_offset, signal_text.len())]
+            .iter()
+            .map(|text| Span::raw(*text))
+            .collect::<Vec<Span<'_>>>();
+        buttons.insert(0, Span::raw(signal_text[0]));
+        buttons[selected - scroll_offset] = Span::styled(
+            signal_text[selected],
+            self.colours.currently_selected_text_style,
+        );
```

Comment:
This method doesn't really seem to work that well if you scroll past the first row and do clicking:

![Peek 2020-10-17 17-07](https://user-images.githubusercontent.com/34804052/96353560-3b8cf180-109b-11eb-959e-42e779b4886a.gif)

While it *does* select the correct entry, the list jumps around and it looks strange.  In comparison, try clicking on an entry in, say, the temperature list or the processes list.  The list should not shift when you click on an entry.

Look at something like `temp_table.rs` at line 51 on how I currently calculate the scroll offset:

```rs
let start_position = get_start_position(
    usize::from(
        (draw_loc.height + (1 - table_gap)).saturating_sub(self.table_height_offset),
    ),
    &temp_widget_state.scroll_state.scroll_direction,
    &mut temp_widget_state.scroll_state.previous_scroll_position,
    temp_widget_state.scroll_state.current_scroll_position,
    app_state.is_force_redraw,
);
//...
let sliced_vec = &temp_sensor_data[start_position..]; // This is properly offset.
```

The gist of it is that I already have a function that manages the calculation of where to start drawing the a scrollable list, and it requires storing some additional fields in the struct to do so.  I would have preferred to make this more intuitive to add, but I don't really have the time to do that right now unfortunately (I was planning on doing it later).  If this is a bit too confusing please let me know, I can try to help a bit more.